bam2bcf.c

```diff
@@ -78,7 +78,7 @@ int bcf_call_glfgen(int _n, const bam_pileup1_t *pl, int ref_base, bcf_callaux_t
     if (bca->max_bases < _n) {
         bca->max_bases = _n;
         kroundup32(bca->max_bases);
-        bca->bases = (uint16_t*)realloc(bca->bases, 2 * bca->max_bases);
+        bca->bases = (uint16_t*)realloc(bca->bases, 2 * (unsigned)bca->max_bases);
```

Comment:
I don't think this is quite the right fix here.  The problem is actually triggered if `kroundup32()` tries to round up a negative value (we fixed it a while back to work with all possible +ve numbers).  The full solution is to change line 76 above to:
```
    if (_n <= 0) return -1;
```
to deal with the unlikely event of negative `_n`, then cast `bca->max_bases` to `size_t` here so the `* 2` can't overflow it on 64-bit platforms.

It would be good to put this change in its own commit, instead of having it with the unrelated regex fix.

Reply:
Okay.